### PR TITLE
Rebase and merge with TurtleArt and other changes related to Port

### LINKED ISSUE #1: Port from GConf to Gio.Settings
https://github.com/sugarlabs/activity-turtle-flags/blob/ac20359849272efa9c1978d979218b5782c21523/collaboration/buddy.py#L98

### LINKED ISSUE #4: Rebase and merge with Turtle Art
Much common code.

### PR DESCRIPTION
**Pull request by Saumya-Mishra9129**
- [x] Rebase and merge with TurtleArt

- [x] Port from GStreamer Gst.Message.structure to get_structure()

- [x] Port to Python3

- [x] Port to GTK3

- [x] Port from GConf to GioSettings

Fixes #4, Fixes #1.
@quozl @pro-panda Please review.

### CODE REVIEW

**Comment by Saumya-Mishra9129** [comment]
Getting this error while testing 

 Screenshot from 2020-06-10 01-20-04

@srevinsaju @chimosky can you review please.

**Comment by srevinsaju** [comment]
@Saumya-Mishra9129 as the error suggests, there is no *.gschema file as there is on turtleart-activity.

**Comment by Saumya-Mishra9129** [comment]
Yeah Thanks, I probably forgot to add that file. I will fix this in morning
😁

**Comment by srevinsaju** [comment]
Ok; I will review the diff tomorrow morning too. (got held up with some work)

File: plugins/camera_sensor/tacamera.py [hunk 1]
```diff
@@ -51,9 +51,9 @@ def __init__(self, device='/dev/video0'):
 
     def _on_message(self, bus, message):
         ''' We get a message if a pixbuf is available '''
-        if message.structure is not None:
-            if message.structure.get_name() == 'pixbuf':
-                self.pixbuf = message.structure['pixbuf']
+        if message.get_structure is not None:
+            if message.get_structure.get_name() == 'pixbuf':
+                self.pixbuf = message.get_structure['pixbuf']
```
Comment (chimosky) [inline plugins/camera_sensor/tacamera.py]: The calls to `message.get_structure` would return the function object and not the structure of the message as intended, it should be `message.get_structure()`.

**Review by chimosky — CHANGES_REQUESTED** [review]

File: turtleflags.py [hunk 2]
```diff
@@ -266,14 +266,14 @@ def _parse_command_line(self):
         try:
             opts, args = getopt.getopt(argv[1:], 'hor',
                                        ['help', 'output_png', 'run'])
-        except getopt.GetoptError, err:
-            print str(err)
-            print self._HELP_MSG
+        except getopt.GetoptError as err:
+            print(str(err))
```
Comment (chimosky) [inline turtleflags.py:294]: You can also print `err.msg` instead of converting err to a string.

**Comment by chimosky** [comment]
Reviewed, thanks.

As this is an update of upstream changes, all your commit messages should reflect that.

File: plugins/camera_sensor/tacamera.py (hunk 1 above)
Comment (Saumya-Mishra9129) [reply plugins/camera_sensor/tacamera.py]: This is probably fixed in ad48974

File: turtleflags.py (hunk 2 above)
Comment (Saumya-Mishra9129) [reply turtleflags.py]: Done in b13917f

**Comment by Saumya-Mishra9129** [comment]
@quozl @chimosky @JuiP @srevinsaju , Please review and test.

**Comment by srevinsaju** [comment]
Ok as I can't do a huge diff review, I will do an activity testing :D

**Comment by srevinsaju** [comment]
@Saumya-Mishra9129 what is the test procedure for this activity? I was not able to click on any of the flags, does it use any key combination to activate them?

**Comment by Saumya-Mishra9129** [comment]
> @Saumya-Mishra9129 what is the test procedure for this activity? I was not able to click on any of the flags, does it use any key combination to activate them?

No, Actually there were some minor bugs in the code. Fixed them. @chimosky @quozl @srevinsaju Please review and merge.

File: collaboration/connectionmanager.py [hunk 3]
```diff
@@ -16,28 +16,35 @@
...
 from dbus import PROPERTIES_IFACE
-from telepathy.interfaces import (ACCOUNT, ACCOUNT_MANAGER)
-from telepathy.constants import CONNECTION_STATUS_CONNECTED
+
+from gi.repository import TelepathyGLib
+
+# from telepathy.interfaces import (ACCOUNT, ACCOUNT_MANAGER)
+ACCOUNT = TelepathyGLib.IFACE_ACCOUNT
+ACCOUNT_MANAGER = TelepathyGLib.IFACE_ACCOUNT_MANAGER
+CONNECTION_STATUS_CONNECTED = TelepathyGLib.ConnectionStatus.CONNECTED
```
Comment (chimosky) [inline collaboration/connectionmanager.py:32]: If this isn't used, there might be no need to add them.

File: turtleflags.py [hunk 4]
```diff
@@ -113,14 +115,37 @@ def __init__(self):
...
+        path = os.path.join(get_path(None, 'data'), 'schemas')
+        if not os.access(path, os.F_OK):
+            os.makedirs(path)
+
+        # create compiled schema file if missing
+        compiled = os.path.join(path, 'gschemas.compiled')
+        if not os.access(compiled, os.R_OK):
+            src = '%s.gschema.xml' % self._GIO_SETTINGS
+            lines = open(os.path.join(activity_root, src), 'r').readlines()
+            open(os.path.join(path, src), 'w').writelines(lines)
```
Comment (chimosky) [inline turtleflags.py:137]: The schema file wasn't closed after writing to it.

File: collaboration/connectionmanager.py (hunk 3 above)
Comment (Saumya-Mishra9129) [reply collaboration/connectionmanager.py:32]: We may also need to update upstream TurtleArt, I have rebased it with TutrleArt.

Comment (Saumya-Mishra9129) [reply collaboration/connectionmanager.py:32]: These constants are used in file later. I dont think it is good to remove now.

File: turtleflags.py (hunk 4 above)
Comment (Saumya-Mishra9129) [reply turtleflags.py]: Done in a016724. @chimosky Please review and merge.

File: collaboration/connectionmanager.py (hunk 3 above)
Comment (chimosky) [reply collaboration/connectionmanager.py:32]: If they're used then there's no reason to remove them.

**Comment by Saumya-Mishra9129** [comment]
@chimosky I have tested activity works fine. Please merge.

**Comment by chimosky** [comment]
Reviewed, thanks. 

Collaboration seems to be broken for the activity, although this PR has nothing to do with it but if you have the time you can fix it in a new PR. Thanks.

**Pull request merged**